### ISSUE TITLE
[hip-runtime-amd] build fails if no devices found

### ISSUE THREAD

**Issue by acxz** [rocm-arch/rocm-arch#762]
```
CMake Error at /build/hip-runtime-amd/src/HIP-rocm-5.1.1/tests/hit/HIT.cmake:51 (string):
  string sub-command LENGTH requires two arguments.
Call Stack (most recent call first):
  CMakeLists.txt:438 (include)


CMake Error at /build/hip-runtime-amd/src/HIP-rocm-5.1.1/tests/hit/HIT.cmake:54 (if):
  if given arguments:

    "GREATER_EQUAL" "1"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:438 (include)


-- Configuring incomplete, errors occurred!
```

Solution has been resolved upstream, but not released: https://github.com/ROCm-Developer-Tools/HIP/commit/d221fb6ebbe24d66855af8160687aa01b0112dec

**Comment by tpkessler** [comment]
Patched with #763.

**Comment by acxz** [comment]
Patch incorporated in new release: https://github.com/rocm-arch/rocm-arch/commit/d49db7f5139aba3953e7c8899d105b0477e92948
